src/unitycatalog/ai/utils.py

```diff
@@ -113,3 +141,148 @@ def validate_full_function_name(function_name: str) -> FullFunctionName:
+
+        types += (FunctionInfo,)
+    except ImportError:
+        pass
+
+    return types
+
+
+@dataclass
+class PydanticField:
+    pydantic_type: Type
+    description: Optional[str] = None
+    default: Optional[Any] = None
+
+
+def param_info_to_pydantic_type(param_info: Any) -> PydanticField:
+    """
+    Convert Unity Catalog function parameter information to Pydantic type.
+
+    Args:
+        param_info: The Unity Catalog function parameter information.
+            It must be either databricks.sdk.service.catalog.FunctionParameterInfo or
+            unitycatalog.types.function_info.InputParamsParameter object.
+    """
+    if not isinstance(param_info, supported_param_info_types()):
+        raise TypeError(f"Unsupported parameter info type: {type(param_info)}")
+    if param_info.type_json is None:
+        raise ValueError(f"Parameter type json is None for parameter {param_info.name}.")
+    type_json = json.loads(param_info.type_json)
+    nullable = type_json.get("nullable")
+    pydantic_type = uc_type_json_to_pydantic_type(type_json["type"])
+    if nullable:
+        pydantic_type = Optional[pydantic_type]
+    default = None
+    description = param_info.comment or ""
+    if param_info.parameter_default:
+        pydantic_type = Optional[pydantic_type]
+        # Note: DEFAULT is supported for LANGUAGE SQL only.
+        # TODO: verify this for all types
+        default = json.loads(param_info.parameter_default)
+        description = f"{description} (Default: {param_info.parameter_default})"
+    return PydanticField(
+        pydantic_type=pydantic_type,
+        description=description,
+        default=default,
+    )
+
+
+def generate_function_input_params_schema(function_info: Any) -> Type[BaseModel]:
+    """
+    Generate a Pydantic model based on a Unity Catalog function information.
+
+    Args:
+        function_info: The Unity Catalog function information.
+            It must either be databricks.sdk.service.catalog.FunctionInfo or
+            unitycatalog.types.function_info.FunctionInfo object.
+
+    Returns:
+        The Pydantic model representing the input parameters of the function.
+    """
+    if not isinstance(function_info, supported_function_info_types()):
+        raise TypeError(f"Unsupported function info type: {type(function_info)}")
+    if function_info.input_params is None:
+        return BaseModel
+    param_infos = function_info.input_params.parameters
+    if param_infos is None:
+        raise ValueError("Function input parameters are None.")
+    fields = {}
+    for param_info in param_infos:
+        pydantic_field = param_info_to_pydantic_type(param_info)
+        fields[param_info.name] = (
+            pydantic_field.pydantic_type,
+            Field(default=pydantic_field.default, description=pydantic_field.description),
+        )
+    return create_model(
+        f"{function_info.catalog_name}__{function_info.schema_name}__{function_info.name}__params",
```

Comment:
Can catalog or schema contain non-ascii characters such as Japanese?

Reply:
It appears they can.

Reply:
What's the concern about non-ascii characters here? I added a test and seems it works fine

Reply:
didn't know non-ascii characters can be used in a class name